pkg/kubelet/kubelet.go

```diff
@@ -2577,21 +2580,25 @@ func (kl *Kubelet) HandlePodAdditions(pods []*v1.Pod) {
 				podCopy := pod.DeepCopy()
 				kl.updateContainerResourceAllocation(podCopy)
 
-				// Check if we can admit the pod; if not, reject it.
-				if ok, reason, message := kl.canAdmitPod(activePods, podCopy); !ok {
-					kl.rejectPod(pod, reason, message)
-					continue
+				if needToAdmit(pod) {
+					// Check if we can admit the pod; if not, reject it.
+					if ok, reason, message := kl.canAdmitPod(activePods, podCopy); !ok {
+						kl.rejectPod(pod, reason, message)
+						continue
+					}
```

Comment:
Regardless of whether the kubelet restart should not affect all cases or not, this change is not desirable.

`canAdmitPod` also includes the eviction process of the kubelet, we cannot disable it for all running pods. Kubelet should be able to evict running pods for various reasons (ephemeral storage pressure, ...).

You first need to define concrete buggy cases and address them one by one.

Reply:
very good point about eviction and the eviction manager

Reply:
Evict running pods won't get in this func,  HandlePodAdditions only handle pods for new created or kubelet restart;
This change will only affect for running pods when kubelet restart.